### PR TITLE
fix(jwt-auth): real_payload was overridden by malicious payload

### LINKED ISSUE #10967: bug: 插件 jwt-auth 的 function get_real_payload 存在 key 和 exp 被恶意修改的可能
### Current Behavior

get_real_payload 中使用了 merge（real_payload, extra_payload）的方式来合并 payload， 但作为参数传入的 extra_payload 并未做严格检查，调用者可以任意传入 key 和 exp ，覆盖 real_payload 中的值。

### Expected Behavior

修改代码，当传入的 payload 不为空时，采用 merge（ extra_payload， real_payload）合并，并返回 extra_payload

### Error Logs

_No response_

### Steps to Reproduce

1. 启用 jwt-auth 插件
2. 通过 public-api插件暴露 /apisix/plugin/jwt/sign 接口
3. 构造 { “key":"letmein", exp: 123456} 作为 DATA 提交至上述接口
4. 检查 response 中的 JWT，可发现 payload 中的值被替换为上述 DATA

### Environment

- APISIX version (run `apisix version`): 3.8.0
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by jaspertian**
### Description

Fixes #10967 

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by Gallardot — CHANGES_REQUESTED** [review]
please refer [jwt-auth.t](https://github.com/apache/apisix/blob/master/t/plugin/jwt-auth.t) and add some UT. Rest LGTM.

**Comment by monkeyDluffy6017** [comment]
Test cases are needed

**Comment by jaspertian** [comment]
> Test cases are needed

Got it.

File: t/plugin/jwt-auth4.t [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            $block->set_value("response_body", "passed\n");
+        }
+    }
+});
+
+run_tests;
+
+__DATA__
+
+=== TEST 24: verify the real_payload's value (key & exp) is not overridden by malicious payload
```
[the remark this answers is not in the document]
Comment (jaspertian) [reply t/plugin/jwt-auth4.t]: Sorry, I got some troube when building local test env. 

I'm trying to fix it and will feedback asap.

[the remark this answers is not in the document]
Comment (jaspertian) [reply t/plugin/jwt-auth4.t]: I re-implement the test case and verified in local test env, please check it.

Any problem , pls contact me. thx!

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]
LGTM! Thank you for your contribution!

**Pull request merged**